### ISSUE TITLE
MacOS Sonoma kbfs finder integration

### ISSUE THREAD

**Issue by marcofranssen** [keybase/client#25951]
Since the release of MacOS Sonoma I expect the finder integration will be broken again on the new MacOS release.

Can we get a similar fix as we got for MacOS Ventura/Monterey in the past?

See here one of those issues:

- https://github.com/keybase/client/issues/24677
- https://github.com/keybase/client/issues/24636
- https://github.com/keybase/client/issues/25043

**Comment by lychnikon** [comment]
This seems to be fixed with an update to kbfuse 4.5; my device received the update automatically today.

**Comment by marcofranssen** [comment]
> This seems to be fixed with an update to kbfuse 4.5; my device received the update automatically today.

So you are running latest macOS already? Can I upgrade safely?

If so, big big kudos to the maintainers for fixing it this quickly. ❤️

**Comment by lychnikon** [comment]
I upgraded to Sonoma and it broke KBFS. Today an update pushed to Keybase that prompted to reauthorize the fbfuse extension. rebooted and it works in Finder and CL again

**Comment by marcofranssen** [comment]
Awesome work 🎉 I have upgraded to Sonoma and everything continued to work as it was.

Thanks and gratitude to the maintainers.

**Comment by steveswing** [comment]
I encountered this again today after the Sonoma 14.3 update.

I checked Gatekeeper with `spctl --assess -vvv --type install /Library/Filesystems/kbfuse.fs` and it reported:

```
/Library/Filesystems/kbfuse.fs: rejected
source=Unnotarized Developer ID
origin=Developer ID Application: Keybase, Inc. (99229SGT5K)
```

Eventually I quit Keybase completely from the hamburger menu and I checked for any `com.apple.quarantine` extended attributes using:

`sudo xattr -r /Applications/Keybase.app /Library/Filesystems/kbfuse.fs`

There were many. I used the following command to remove them:

`sudo xattr -rd com.apple.quarantine /Applications/Keybase.app /Library/Filesystems/kbfuse.fs`

I didn't try this but if you use `xattr -d com.apple.quarantine Keybase-arm64.dmg` to remove the extended attribute before you open the dmg, it may install Keybase.app without the extended attributes and avoid triggering Gatekeeper scans.
